process/peer/process_test.go

```
@@ -2951,3 +2952,83 @@ func TestValidatorStatisticsProcessor_getActualList(t *testing.T) {
 	computedJailedList := peer.GetActualList(jailedPeer)
 	assert.Equal(t, jailedList, computedJailedList)
 }
+
+func TestValidatorStatistics_SearchInMap(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work when found in cache", func(t *testing.T) {
+		t.Parallel()
+
+		cache := createMockCache()
+
+		_, header := generateTestShardBlockHeaders(cache)
+		headerHash := []byte("headerHash")
+		cache[string(headerHash)] = header
+
+		arguments := createMockArguments()
+
+		validatorStatistics, _ := peer.NewValidatorStatisticsProcessor(arguments)
+
+		retHeader, err := validatorStatistics.SearchInMap(headerHash, cache)
+		require.Nil(t, err)
+		require.Equal(t, header, retHeader)
+	})
+
+	t.Run("should work when found in headers pool", func(t *testing.T) {
+		t.Parallel()
+
+		cache := createMockCache()
+
+		_, header := generateTestShardBlockHeaders(cache)
```

Can we also "compress" lines in these tests:
- `should fail when not found in headers pool`
- `should work when found in headers pool`

---

i think it's ok for better visibility